sql_builder.go

```diff
@@ -27,6 +27,17 @@ func newSQLBuilder(q Query, m *Model, addColumns ...string) *sqlBuilder {
 	}
 }
 
+func hasLimitOrOffset(sqlString string) bool {
+	if matched, _ := regexp.MatchString("(?i).* limit [0-9]*( ?, ?[0-9]*)?$", sqlString); matched {
+		return matched
+	}
+
+	if matched, _ := regexp.MatchString("(?i).* offset [0-9]*$", sqlString); matched {
```

Comment:
If you ask my opinion, i would change 2 things.
1. The first number should be mandatory, according to syntax validity. [0-9]+
2. I wouldn't use just regular space in a regex because there could be a new line or a tab there.
 Something like - ((?i).*[ \t]*limit[ \t]*[0-9]+([ \t]*[,][ \t]*[0-9]*)?$)

[ \t]* - catches whitespace, tab and new line.
https://regex101.com/r/sW91rT/1

Reply:
It will also match subqueries, I'm not sure that's a good idea. It will not work with the `FETCH FIRST x ROWS ONLY` syntax too.

How about not trying to rewrite the query, and ask the user to not use such clauses when using automatic pagination?

Reply:
with $ in the end, it will not match sub queries

Reply:
i will whip out a test case playground, ive done it previously but i accidently deletes the shared bookmark and booked marked without share url..